test/fail_compilation/fail15535.d

```diff
@@ -0,0 +1,19 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/fail15535.d(14): Error: goto default not allowed in final switch statement
+---
+*/
+
+void test()
+{
+    int i;
+    switch (i) {
+        case 0:
+            final switch(i) {
+                case 1: goto default;
```

Comment:
Yeah, I wasn't sure about keeping this in.
I left the outer switch in, so as to demonstrate/test that the "goto default" does not get accepted because of an outer "default:" case.